### PR TITLE
Add an annotation option to opt-in autopilot.

### PR DESCRIPTION
**Pull request by incfly**
https://github.com/istio/istio/issues/17276

### CODE REVIEW

File: annotation/annotations.yaml [hunk 1]
```diff
@@ -74,6 +74,11 @@ annotations:
       istio-proxy sidecar.
     deprecated: false
     hidden: false
+  - name: security.istio.io/OptInAutoMTLS
+    description: Determined whether the client Envoy opt-in auto mTLS behavior. This overrides
```
Comment (linsun) [inline annotation/annotations.yaml]: s/Determined/Determine/?

File: annotation/annotations.yaml [hunk 2]
```diff
@@ -74,6 +74,11 @@ annotations:
       istio-proxy sidecar.
     deprecated: false
     hidden: false
+  - name: security.istio.io/OptInAutoMTLS
```
Comment (linsun) [inline annotation/annotations.yaml]: what do you think of just calling it security.istio.io/autoMTLS

File: annotation/annotations.yaml (hunk 1 above)
Comment (incfly) [reply annotation/annotations.yaml]: done.

Comment (smawson) [reply annotation/annotations.yaml]: Determines whether the client proxy uses auto mTLS. This overrides the mesh default specified in MeshConfig.enable_auto-mtls.

File: annotation/annotations.yaml (hunk 2 above)
Comment (incfly) [reply annotation/annotations.yaml]: Sounds good. Done.

Just to confirm, the value pairs is still `true/false`. right?

File: annotation/annotations.yaml (hunk 1 above)
Comment (incfly) [reply annotation/annotations.yaml]: Done. reworded.

**Comment by incfly** [comment]
Thanks for the comments. Updated, PTAL.

@smawson @linsun

File: annotation/annotations.yaml [hunk 3]
```diff
@@ -74,6 +74,11 @@ annotations:
       istio-proxy sidecar.
     deprecated: false
     hidden: false
+  - name: security.istio.io/autoMTLS
```
Comment (smawson) [inline annotation/annotations.yaml]: Can you move this so it is not in the middle of the sidecar annotations? Basically keep the annotations sorted in this file would be helpful.

File: annotation/annotations.yaml (hunk 2 above)
Comment (linsun) [reply annotation/annotations.yaml]: yes +1

File: annotation/annotations.yaml (hunk 3 above)
Comment (incfly) [reply annotation/annotations.yaml]: Done.

**Comment by incfly** [comment]
@linsun  @smawson 
Mind taking another look to see if this is ready to be merged? Thanks!

**Pull request merged**